lib/plug/logger.ex

```diff
@@ -36,12 +43,17 @@ defmodule Plug.Logger do
   @behaviour Plug
 
   def init(opts) do
-    Keyword.get(opts, :log, :info)
+    {
+      :level, Keyword.get(opts, :log, :info),
+      :req_id_header, Keyword.get(opts, :request_id_http_header, "x-request-id")
+    }
   end
 
-  def call(conn, level) do
-    request_id = external_request_id(conn) || generate_request_id()
-    Logger.metadata(request_id: request_id)
+  def call(conn, {:level, level, :req_id_header, req_id_header}) do
+
+    if(request_id = header(conn, req_id_header)) do
+      Logger.metadata(request_id: request_id)
```

Comment:
I.e. if I remove this Logger in favor of my own logging, i still want it to just work and not to any request id setup.